From 755fbbeb1493ea8814f570760f9a51ae0538d32f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=A9r=C3=A9mie=20Galarneau?=
 <jeremie.galarneau@efficios.com>
Date: Wed, 9 Nov 2022 13:59:50 -0500
Subject: [PATCH 03/23] Add "_time32" suffixed variants of syscalls to
 table-syscall-inout.txt
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

In addressing a number of y2k38 upstream bugs, some syscalls had a
64-bit time variant added. In doing so, the superseded syscalls were
renamed *_time32.

table-syscall-inout.txt is adapted to consider both variants of these
syscalls.

This change is lifted from
https://review.lttng.org/c/lttng-modules/+/3208, originally produced by
Michael Jeanson.

Upstream-Status: Backport

Signed-off-by: Jérémie Galarneau <jeremie.galarneau@efficios.com>
Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Change-Id: I34511fb80fea1d17fd5eea7fccaffe1b5335a822
---
 .../syscalls/table-syscall-inout.txt          | 28 +++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/include/instrumentation/syscalls/table-syscall-inout.txt b/include/instrumentation/syscalls/table-syscall-inout.txt
index cb835a52..89759a3c 100644
--- a/include/instrumentation/syscalls/table-syscall-inout.txt
+++ b/include/instrumentation/syscalls/table-syscall-inout.txt
@@ -33,6 +33,7 @@ syscall dup nbargs 1 rw: (r)
 syscall dup2 nbargs 2 rw: (r, r)
 syscall pause nbargs 0
 syscall nanosleep nbargs 2 rw: (r, w)
+syscall nanosleep_time32 nbargs 2 rw: (r, w)
 syscall getitimer nbargs 2 rw: (r, w)
 syscall alarm nbargs 1 rw: (r)
 syscall setitimer nbargs 3 rw: (r, r, w)
@@ -70,7 +71,9 @@ syscall flock nbargs 2 rw: (r, r)
 syscall fsync nbargs 1 rw: (r)
 syscall fdatasync nbargs 1 rw: (r)
 syscall truncate nbargs 2 rw: (r, r)
+syscall truncate64 nbargs 2 rw: (r, r)
 syscall ftruncate nbargs 2 rw: (r, r)
+syscall ftruncate64 nbargs 2 rw: (r, r)
 syscall getdents nbargs 3 rw: (r, w, r)
 syscall getcwd nbargs 2 rw: (w, r)
 syscall chdir nbargs 1 rw: (r)
@@ -122,10 +125,12 @@ syscall capget nbargs 2 rw: (r, w)
 syscall capset nbargs 2 rw: (r, r)
 syscall rt_sigpending nbargs 2 rw: (w, r)
 syscall rt_sigtimedwait nbargs 4 rw: (w, w, r, r)
+syscall rt_sigtimedwait_time32 nbargs 4 rw: (w, w, r, r)
 syscall rt_sigqueueinfo nbargs 3 rw: (r, r, r)
 syscall rt_sigsuspend nbargs 2 rw: (r, r)
 syscall sigaltstack nbargs 2 rw: (r, rw)
 syscall utime nbargs 2 rw: (r, r)
+syscall utime32 nbargs 2 rw: (r, r)
 syscall mknod nbargs 3 rw: (r, r, r)
 syscall personality nbargs 1 rw: (r)
 syscall ustat nbargs 2 rw: (r, w)
@@ -141,6 +146,7 @@ syscall sched_getscheduler nbargs 1 rw: (r)
 syscall sched_get_priority_max nbargs 1 rw: (r)
 syscall sched_get_priority_min nbargs 1 rw: (r)
 syscall sched_rr_get_interval nbargs 2 rw: (r, w)
+syscall sched_rr_get_interval_time32 nbargs 2 rw: (r, w)
 syscall mlock nbargs 2 rw: (r, r)
 syscall munlock nbargs 2 rw: (r, r)
 syscall mlockall nbargs 1 rw: (r)
@@ -150,6 +156,7 @@ syscall pivot_root nbargs 2 rw: (r, r)
 syscall sysctl nbargs 1 rw: (rw)
 syscall prctl nbargs 5 rw: (r, rw, r, r, r)
 syscall adjtimex nbargs 1 rw: (rw)
+syscall adjtimex_time32 nbargs 1 rw: (rw)
 syscall setrlimit nbargs 2 rw: (r, r)
 syscall chroot nbargs 1 rw: (r)
 syscall sync nbargs 0
@@ -181,12 +188,15 @@ syscall lremovexattr nbargs 2 rw: (r, r)
 syscall fremovexattr nbargs 2 rw: (r, r)
 syscall tkill nbargs 2 rw: (r, r)
 syscall time nbargs 1 rw: (w)
+syscall time32 nbargs 1 rw: (w)
 syscall futex nbargs 6 rw: (rw, r, r, r, rw, r)
+syscall futex_time32 nbargs 6 rw: (rw, r, r, r, rw, r)
 syscall sched_setaffinity nbargs 3 rw: (r, r, r)
 syscall sched_getaffinity nbargs 3 rw: (r, r, w)
 syscall io_setup nbargs 2 rw: (r, r)
 syscall io_destroy nbargs 1 rw: (r)
 syscall io_getevents nbargs 5 rw: (r, r, r, w, rw)
+syscall io_getevents_time32 nbargs 5 rw: (r, r, r, w, rw)
 syscall io_submit nbargs 3 rw: (r, r, r)
 syscall io_cancel nbargs 3 rw: (r, r, w)
 syscall lookup_dcookie nbargs 3 rw: (r, w, r)
@@ -197,27 +207,37 @@ syscall set_tid_address nbargs 1 rw: (r)
 syscall restart_syscall nbargs 0
 syscall semtimedop nbargs 4 rw: (r, r, r, rw)
 syscall fadvise64 nbargs 4 rw: (r, r, r, r)
+syscall fadvise64_64 nbargs 4 rw: (r, r, r, r)
 syscall timer_create nbargs 3 rw: (r, r, w)
 syscall timer_settime nbargs 4 rw: (r, r, r, w)
+syscall timer_settime32 nbargs 4 rw: (r, r, r, w)
 syscall timer_gettime nbargs 2 rw: (r, w)
+syscall timer_gettime32 nbargs 2 rw: (r, w)
 syscall timer_getoverrun nbargs 1 rw: (r)
 syscall timer_delete nbargs 1 rw: (r)
 syscall clock_settime nbargs 2 rw: (r, r)
+syscall clock_settime32 nbargs 2 rw: (r, r)
 syscall clock_gettime nbargs 2 rw: (r, w)
+syscall clock_gettime32 nbargs 2 rw: (r, w)
 syscall clock_getres nbargs 2 rw: (r, w)
+syscall clock_getres_time32 nbargs 2 rw: (r, w)
 syscall clock_nanosleep nbargs 4 rw: (r, r, r, w)
+syscall clock_nanosleep_time32 nbargs 4 rw: (r, r, r, w)
 syscall exit_group nbargs 1 rw: (r)
 syscall epoll_wait nbargs 4 rw: (r, w, r, r)
 syscall epoll_ctl nbargs 4 rw: (r, r, r, r)
 syscall tgkill nbargs 3 rw: (r, r, r)
 syscall utimes nbargs 2 rw: (r, r)
+syscall utimes_time32 nbargs 2 rw: (r, r)
 syscall mbind nbargs 6 rw: (r, r, r, r, r, r)
 syscall set_mempolicy nbargs 3 rw: (r, r, r)
 syscall get_mempolicy nbargs 5 rw: (w, w, r, r, r)
 syscall mq_open nbargs 4 rw: (r, r, r, r)
 syscall mq_unlink nbargs 1 rw: (r)
 syscall mq_timedsend nbargs 5 rw: (r, r, r, r, r)
+syscall mq_timedsend_time32 nbargs 5 rw: (r, r, r, r, r)
 syscall mq_timedreceive nbargs 5 rw: (r, w, r, w, r)
+syscall mq_timedreceive_time32 nbargs 5 rw: (r, w, r, w, r)
 syscall mq_notify nbargs 2 rw: (r, r)
 syscall mq_getsetattr nbargs 3 rw: (r, r, w)
 syscall kexec_load nbargs 4 rw: (r, r, r, r)
@@ -236,6 +256,7 @@ syscall mkdirat nbargs 3 rw: (r, r, r)
 syscall mknodat nbargs 4 rw: (r, r, r, r)
 syscall fchownat nbargs 5 rw: (r, r, r, r, r)
 syscall futimesat nbargs 3 rw: (r, r, r)
+syscall futimesat_time32 nbargs 3 rw: (r, r, r)
 syscall newfstatat nbargs 4 rw: (r, r, w, r)
 syscall unlinkat nbargs 3 rw: (r, r, r)
 syscall renameat nbargs 4 rw: (r, r, r, r)
@@ -245,7 +266,9 @@ syscall readlinkat nbargs 4 rw: (r, r, w, r)
 syscall fchmodat nbargs 3 rw: (r, r, r)
 syscall faccessat nbargs 3 rw: (r, r, r)
 syscall pselect6 nbargs 6 rw: (r, rw, rw, rw, rw, r)
+syscall pselect6_time32 nbargs 6 rw: (r, rw, rw, rw, rw, r)
 syscall ppoll nbargs 5 rw: (rw, r, r, r, r)
+syscall ppoll_time32 nbargs 5 rw: (rw, r, r, r, r)
 syscall unshare nbargs 1 rw: (r)
 syscall set_robust_list nbargs 2 rw: (r, r)
 syscall get_robust_list nbargs 3 rw: (r, w, w)
@@ -255,13 +278,16 @@ syscall sync_file_range nbargs 4 rw: (r, r, r, r)
 syscall vmsplice nbargs 4 rw: (r, r, r, r)
 syscall move_pages nbargs 6 rw: (r, r, r, r, w, r)
 syscall utimensat nbargs 4 rw: (r, r, r, r)
+syscall utimensat_time32 nbargs 4 rw: (r, r, r, r)
 syscall epoll_pwait nbargs 6 rw: (r, w, r, r, r, r)
 syscall signalfd nbargs 3 rw: (r, r, r)
 syscall timerfd_create nbargs 2 rw: (r, r)
 syscall eventfd nbargs 1 rw: (r)
 syscall fallocate nbargs 4 rw: (r, r, r, r)
 syscall timerfd_settime nbargs 4 rw: (r, r, r, w)
+syscall timerfd_settime32 nbargs 4 rw: (r, r, r, w)
 syscall timerfd_gettime nbargs 2 rw: (r, w)
+syscall timerfd_gettime32 nbargs 2 rw: (r, w)
 syscall accept4 nbargs 4 rw: (r, w, rw, r)
 syscall signalfd4 nbargs 4 rw: (r, r, r, r)
 syscall eventfd2 nbargs 2 rw: (r, r)
@@ -274,12 +300,14 @@ syscall pwritev nbargs 5 rw: (r, r, r, r, r)
 syscall rt_tgsigqueueinfo nbargs 4 rw: (r, r, r, r)
 syscall perf_event_open nbargs 5 rw: (r, r, r, r, r)
 syscall recvmmsg nbargs 5 rw: (r, w, r, r, rw)
+syscall recvmmsg_time32 nbargs 5 rw: (r, w, r, r, rw)
 syscall fanotify_init nbargs 2 rw: (r, r)
 syscall fanotify_mark nbargs 5 rw: (r, r, r, r, r)
 syscall prlimit64 nbargs 4 rw: (r, r, r, w)
 syscall name_to_handle_at nbargs 5 rw: (r, r, rw, w, r)
 syscall open_by_handle_at nbargs 3 rw: (r, r, r)
 syscall clock_adjtime nbargs 2 rw: (r, rw)
+syscall clock_adjtime32 nbargs 2 rw: (r, rw)
 syscall syncfs nbargs 1 rw: (r)
 syscall sendmmsg nbargs 4 rw: (r, r, r, r)
 syscall setns nbargs 2 rw: (r, r)
-- 
2.34.1

